src/init.cpp

```diff
@@ -1444,8 +1445,18 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     nTotalCache = std::max(nTotalCache, nMinDbCache << 20); // total cache cannot be less than nMinDbCache
     nTotalCache = std::min(nTotalCache, nMaxDbCache << 20); // total cache cannot be greated than nMaxDbcache
     int64_t nBlockTreeDBCache = nTotalCache / 8;
-    if (nBlockTreeDBCache > (1 << 21) && !GetBoolArg("-txindex", false))
-        nBlockTreeDBCache = (1 << 21); // block tree db cache shouldn't be larger than 2 MiB
+    if (GetBoolArg("-addressindex", false)) {
+        if (!GetBoolArg("-txindex", false)) {
+            return InitError(_("-addressindex requires -txindex."));
+        }
+        // enable 3/4 of the cache if addressindex is enabled
+        nBlockTreeDBCache = nTotalCache * 3 / 4;
+
+    } else {
```

Comment:
We can simplify this with just a single `else if` block, but minor nit

Reply:
To reduce the diff size, you could leave the two lines removed as they originally were and not have the else { } block.  If addressindex is set, nBlockTreeDBCache will get set.

Reply:
Just for future reference, nothing is too much of a nit for me; please always provide them. I love these kinds of improvements.